pgx.opam

```diff
@@ -19,7 +19,7 @@ depends: [
   "re"
   "sexplib0" {>= "v0.11.0"}
 
-  "bisect_ppx" {build & >= "1.4.1"}
+  "bisect_ppx" {dev & >= "2.0.0"}
```

Comment:
Where is `dev` documented?

Reply:
I saw it first on the bisect_ppx repo https://github.com/aantron/bisect_ppx#dune
opam seems to support it -> https://opam.ocaml.org/doc/Manual.html#Package-variables

Reply:
Hmm I think it's saying something different than build though:

> dev - true if this is a development package, i.e. it was not built from a release archive

vs. 

> build - limits the dependency to a build-time one, avoiding recompilation if it changes

I think we still want this to be `build`?

Reply:
Hm although I guess we can just follow upstream's instructions

Reply:
Hmm i guess for this to truly work for a opam release we'd also need to do the following. Or we can just add bisect as a regular dependency and not change anything. The code will only be instrumented when the BISECT_ENABLE flag is set

```
(* -*- tuareg -*- *)

let preprocess =
  match Sys.getenv "BISECT_ENABLE" with
  | "yes" -> "(preprocess (pps bisect_ppx))"
  | _ -> ""
  | exception Not_found -> ""

let () = Jbuild_plugin.V1.send @@ {|

(library
 (public_name my_lib)
 |} ^ preprocess ^ {|)

|}
```

Reply:
It should be a dev dependency with the instructions posted by @anuragsoni. A build dependency is installed by opam for users of your release in order to build it. Bisect_ppx should not be a dependency of your release at all, and is not necessary to build it. It also should not be a test dependency, because Bisect_ppx is not necessary to build or run tests of your release. It only comes into play when you manually want to run tests with coverage, which normally only happens during development.

Reply:
Thanks for confirming @aantron . I've made a new PR that makes these changes for pgx.